### PR TITLE
refactor(build-infrastructure): Make package manager install command an array of args

### PR DESCRIPTION
**Pull request by tylerbutler**
The package manager's install command was a string, but an array of args is better since we were splitting the string anyway.

### CODE REVIEW

File: build-tools/packages/build-infrastructure/src/packageManagers.ts [hunk 1]
```diff
@@ -35,22 +35,21 @@ export class PackageManager implements IPackageManager {
 		}
 	}
 
-	public installCommand(updateLockfile: boolean): string {
+	public installCommandArgs(updateLockfile: boolean): string[] {
```
Comment (alexvy86) [inline build-tools/packages/build-infrastructure/src/packageManagers.ts:38]: Thoughts? My first read of the other name made me think `install` shouldn't be part of the return value.

Also, the yarn case seems to be incorrect, it should still return "install", right?
```suggestion
	public installCommandWithArgs(updateLockfile: boolean): string[] {
```

File: build-tools/packages/build-infrastructure/src/types.ts [hunk 2]
```diff
@@ -289,14 +289,17 @@ export interface IPackageManager {
 	readonly lockfileName: string;
 
 	/**
-	 * Returns an install command that can be used to install dependencies using this package manager.
+	 * Returns an array of arguments that can be used to install dependencies using this package manager.
```
Comment (alexvy86) [inline build-tools/packages/build-infrastructure/src/types.ts:292]:
```suggestion
	 * Returns an array of arguments (including the name of the command, e.g. "install") that can be used to install dependencies using this package manager.
```

File: build-tools/packages/build-infrastructure/src/types.ts [hunk 3]
```diff
@@ -289,14 +289,17 @@ export interface IPackageManager {
...
 	 *
 	 * @param updateLockfile - If `true`, then the returned command will include flags or arguments necessary to update
 	 * the lockfile during install. If `false`, such flags or arguments should be omitted. Note that the command will
 	 * _not_ include the package manager name istself. For example, the `npm` package manager will return the string
 	 * `"install"`, not `"npm install"`.
+	 *
+	 * @example
+	 * For the pnpm package manager, calling `installCommand(true)` would return `["install", "--no-frozen-lockfile"]`.
```
Comment (alexvy86) [inline build-tools/packages/build-infrastructure/src/types.ts:300]: Or update if going with my rename suggestion.
```suggestion
	 * For the pnpm package manager, calling `installCommandArgs(true)` would return `["install", "--no-frozen-lockfile"]`.
```

File: build-tools/packages/build-infrastructure/src/packageManagers.ts (hunk 1 above)
Comment (tylerbutler) [reply build-tools/packages/build-infrastructure/src/packageManagers.ts:38]: > Also, the yarn case seems to be incorrect, it should still return "install", right?

Yeah, incomplete refactor.

**Review by alexvy86 — APPROVED** [review]

**Pull request merged**